clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackOrchestrationV1Provider.groovy

```diff
@@ -115,4 +116,41 @@ class OpenstackOrchestrationV1Provider implements OpenstackOrchestrationProvider
       it.type == "OS::Nova::Server" ? it.physicalResourceId : null
     }
   }
+
+  @Override
+  Resource getInstanceResourceForStack(String region, Stack stack, String instanceName) {
+    List<? extends Resource> resources = handleRequest {
+      //this means it has the ability to list resources 10 levels deep in the heat template hierarchy.
+      //provide a depth of 10 for insurance - the default template has a depth of 4.
+      getRegionClient(region).heat().resources().list(stack.name, 10)
+    }
+    List<String> parts = instanceName.split("-")?.toList()
```

Comment:
Global comment in this class -- Use single quotes when not resolving variables. ('-')